plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidViewHolderRefactoring.java

```diff
@@ -0,0 +1,311 @@
+            return VISIT_SUBTREE;
+        }
+
+        public boolean isInflateInsideIf() {
+            if (this.viewAssignmentStatement != null) {
+                if (getAncestorOrNull(this.viewAssignmentStatement, IfStatement.class) != null) {
+                    return true;
+                } else if (getAncestorOrNull(this.viewAssignmentStatement, SwitchStatement.class) != null) {
+                    return true;
+                } else {
+                    // check whether inflate is inside a conditional assignment
+                    Expression inflateExpression = this.getInflateExpression();
+                    if (inflateExpression != null
+                            && inflateExpression.getNodeType() == ASTNode.CONDITIONAL_EXPRESSION) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        public Expression getInflateExpression() {
+            if (this.viewVariableDeclarationFragment != null) {
+                return this.viewVariableDeclarationFragment.getInitializer();
+            } else if (this.viewVariableAssignment != null) {
+                return this.viewVariableAssignment.getRightHandSide();
+            }
+            return null;
+        }
+
+    }
+
+    static class FindViewByIdVisitor extends ASTVisitor {
+        private List<FindViewByIdItem> items = new ArrayList<FindViewByIdItem>();
+
+        public List<FindViewByIdItem> getItems() {
+            return items;
+        }
+
+        static class FindViewByIdItem {
+            SimpleName variable;
+            Expression findViewByIdExpression;
+            Statement findViewByIdAssignment;
+            VariableDeclarationFragment findViewByIdDeclarationFragment;
+            Assignment findViewByIdVariableAssignment;
+            MethodInvocation findViewByIdInvocation;
+
+            FindViewByIdItem(MethodInvocation node) {
+                this.setAssignment(node);
+            }
+
+            public void setAssignment(MethodInvocation node) {
+                this.findViewByIdInvocation = node;
+                this.findViewByIdDeclarationFragment = getAncestorOrNull(node,
+                        VariableDeclarationFragment.class);
+                if (this.findViewByIdDeclarationFragment != null) {
+                    this.variable = this.findViewByIdDeclarationFragment.getName();
+                    this.findViewByIdAssignment = getAncestor(this.findViewByIdDeclarationFragment,
+                            VariableDeclarationStatement.class);
+                    this.findViewByIdExpression = this.findViewByIdDeclarationFragment.getInitializer();
+                } else {
+                    this.findViewByIdVariableAssignment = getAncestorOrNull(node, Assignment.class);
+                    if (this.findViewByIdVariableAssignment != null) {
+                        this.variable = (SimpleName) this.findViewByIdVariableAssignment.getLeftHandSide();
```

Comment:
Potential `ClassCastException` here.
It could be a `QualifiedName` or a `FieldAcces`, or an `ArrayAccess`, etc...